Non compilable java code generated

**Describe the bug**
* Spring dependency in quarkus code `RequestParam` in REST resource. Should this be `QueryParam` ?
* `JsonIgnore` (whithout import) on generated entity class


**To Reproduce**
Steps to reproduce the behavior. Use the following jdl file:

```
application {
  config {
    baseName productstructure
    packageName nl.gdn.psa
    applicationType monolith
    clientFramework angular
	packageName nl.gdn.psa
  }
  entities Portfolio, Product
}

entity Portfolio{
  name String
}

entity Product {
  name String
  type ProductType
  department String
  contact String
  functionalOwner String
  productOwner String
  description String
  reference String
}

relationship OneToMany {
  Product{parent} to Product
}

relationship OneToOne {
  Portfolio{product} to Product
}

enum ProductType {
  COMPOSITE,
  INFORMATION_PRODUCT,
  SOFTWARE
}
```

`./mvnw (on java 17) `

produces
```
[ERROR] COMPILATION ERROR :
[INFO] -------------------------------------------------------------
[ERROR] source/product-structure/src/main/java/nl/gdn/psa/domain/Product.java:[51,6] cannot find symbol
  symbol:   class JsonIgnore
  location: class nl.gdn.psa.domain.Product
[ERROR] source/product-structure/src/main/java/nl/gdn/psa/web/rest/ProductResource.java:[100,42] cannot find symbol
  symbol:   class RequestParam
  location: class nl.gdn.psa.web.rest.ProductResource
```

If you decide to generate a `ServiceImpl` and DTO on  `Product` there will be as well one missing method on the service: `public List<ProductDTO> findAllWherePortfolioIsNull()`.  Also the `JsonIgnore` is there on the entity, while the entity is exposed via a DTO.

**Expected behavior**
working code?

I tried reproducing this issue with the `main` branch and there is a compilation failure:

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.13.0:compile (default-compile) on project productstructure: Compilation failure
[ERROR] /Users/mraible/Downloads/quarkus/src/main/java/nl/gdn/psa/domain/Product.java:[55,6] cannot find symbol
[ERROR]   symbol:   class JsonIgnore
[ERROR]   location: class nl.gdn.psa.domain.Product
```

--- a/README.md
+++ b/README.md
@@ -106,7 +106,7 @@ During development of blueprint, please note the below steps. They are very impo
 
 # License
 
-Apache-2.0
+Apache 2.0, see [LICENSE](LICENSE).
 
 [npm-image]: https://img.shields.io/npm/v/generator-jhipster-quarkus.svg
 [npm-url]: https://npmjs.org/package/generator-jhipster-quarkus

--- a/generators/quarkus/templates/src/main/java/_package_/_entityPackage_/domain/_persistClass_.java.ejs
+++ b/generators/quarkus/templates/src/main/java/_package_/_entityPackage_/domain/_persistClass_.java.ejs
@@ -380,7 +380,7 @@ public class <%= persistClass %> extends <% if (databaseType === 'sql') { %>Pana
             <%_ } _%>
         <%_ } else { _%>
     @OneToOne(mappedBy = "<%= otherEntityRelationshipName %>")
-    @JsonIgnore
+    <%_ if (importJsonIgnore === true) { _%>@JsonIgnore<%_ } _%>
         <%_ }
         } else if (databaseType === 'mongodb' || databaseType === 'couchbase') {
             if (databaseType === 'mongodb') { _%>

